test/integration/targets/route53/tasks/main.yml

```diff
@@ -21,8 +21,23 @@
       comment: Created in Ansible test {{ resource_prefix }}
     register: z1
 
-  - debug: msg='TODO write tests'
-  - debug: var=z1
+  - assert:
+      that:
+        - z1 is success
+        - z1 is changed
+        - "z1.comment == 'Created in Ansible test {{ resource_prefix }}'"
+
+  - name: List hosted zones
+    route53_info:
+      query: hosted_zone
+      hosted_zone_id: '{{ z1.zone_id }}'
+      max_items: 50
```

Comment:
If you made a couple hosted zones and changed this to a small number you could test next_marker too. Not a blocker though, looks good as-is.